consumer/consumer.go

```diff
@@ -2,23 +2,23 @@ package consumer
 
 import (
 	"context"
-	"github.com/goat-project/goat-proto-go"
+	"github.com/goat-project/goat/consumer/wrapper"
 )
 
 // IPConsumer processes IpRecord-s
 type IPConsumer interface {
 	// ConsumeIps processes all ip records from specified channel and specified client id
-	ConsumeIps(ctx context.Context, id string, ips <-chan goat_grpc.IpRecord) (ResultsChannel, error)
+	ConsumeIps(ctx context.Context, id string, ips <-chan wrapper.RecordWrapper) (ResultsChannel, error)
```

Comment:
Is there a need for three different `*Consumer` interfaces? Since those methods take `RecordWrapper` their signatures are the same.

Reply:
I could merge them into a single interface, but that would mean we can't have a consumer that has consume methods for all of them. That could be useful e.g. if someone decided they need to put VMs and storage records in the same file. If they are separate, the consumers will have to communicate on their own. This way, it can be a single consumer and use a shared state to communicate between the goroutines, which might be useful.

Reply:
I read it 5 times and to be honest I still don't understand :smile: Can you maybe go into more datails, maybe with examples? Or we can discuss it F2F next week.

Reply:
I'll try to explain one more time :smile:
Let's merge the interfaces into something like:
```go
type Consumer interface {
        Consume(ctx context.Context, id string, records <-chan wrapper.RecordWrapper) (ResultsChannel, error)
}
```
Suppose you want to have a consumer that processes IPs, VMs and Storage records. This consumer may only have single `Consume` method which will be called regardless of the record type. There is no way to have separate methods for IPs, VMs and Storages. There is just one.

Reply:
But isn't that the point of a consumer? Process records the same way regardless of their type? That's why we came up with those wrappers.

Reply:
You're right, I'm overthinking this too much. I'll merge them into a single interface.